user/user.go

```diff
@@ -51,17 +54,58 @@ func Create(response http.ResponseWriter, request *http.Request) {
 }
 
 // helper functions perform CRUD operations on user
-// func FindUserByID(response http.ResponseWriter, request *http.Request) {
-// 	user := &User{}
-// 	collectionName := "users"
-// 	objID, _ := primitive.ObjectIDFromHex(id)
-// 	collection := utils.GetCollection(collectionName)
-// 	res := collection.FindOne(ctx, bson.M{"_id": objID})
-// 	if err := res.Decode(user); err != nil {
-// 		return nil, err
-// 	}
-// 	return user, nil
-// }
+func FindUserByID(response http.ResponseWriter, request *http.Request) {
+	// Find a user by user ID
+	response.Header().Set("content-type", "application/json")
+
+	collectionName := "users"
+	userID := mux.Vars(request)["id"]
+	objID, _ := primitive.ObjectIDFromHex(userID)
+
+	res, err := utils.GetMongoDbDoc(collectionName, bson.M{"_id": objID})
+	if err != nil {
+		utils.GetError(err, http.StatusInternalServerError, response)
+		return
+	}
+	utils.GetSuccess("User retrieved successfully", res, response)
+
+}
+
+func UpdateUser(response http.ResponseWriter, request *http.Request) {
+	// Update a user of a given ID
+	response.Header().Set("content-type", "application/json")
+
+	collectionName := "users"
+	userID := mux.Vars(request)["id"]
+	objID, _ := primitive.ObjectIDFromHex(userID)
+
+	res, err := utils.GetMongoDbDoc(collectionName, bson.M{"_id": objID})
+	if err != nil {
+		utils.GetError(err, http.StatusInternalServerError, response)
+		return
+	}
+
+	if res != nil {
+		// 2. Get user fields to be updated from request body
+		var body map[string]interface{}
```

Comment:
Please confirm that this doesn't replace the user document with any random object. Eg test with a Json that has nothing to do with a user schema e.g {"school" :"unilag"}

Reply:
Currently working on this.

Reply:
I have made some updates to this but still have the problem as earlier. Uncertain how to go about this. If there are any pointers you can give I will be grateful.